Sources/HttpPipeline/NIO.swift

```diff
@@ -74,11 +74,16 @@ private final class Handler: ChannelInboundHandler {
       }
     case .end:
       guard let req = self.request else {
-        _ = ctx.channel.write(HTTPServerResponsePart.head(HTTPResponseHead(
-          version: .init(major: 1, minor: 1),
-          status: .init(statusCode: 307),
-          headers: .init([("location", self.baseUrl.absoluteString)])
-        )))
+        ctx.channel.write(
+          HTTPServerResponsePart.head(
+            HTTPResponseHead(
+              version: .init(major: 1, minor: 1),
+              status: .init(statusCode: 307),
+              headers: .init([("location", self.baseUrl.absoluteString)])
+            )
+          ),
+          promise: nil
```

Comment:
@weissi Is it safe to assume that we should favor these `promise: nil` overloads over `_ =` wherever we're not `then`-chaining?

Reply:
yes, there's no difference apart from allocating the promise/future that you'd then throw away with `_ = `. Therefore if you don't need the result, just passing `promise: nil` is just a bit cheaper. Here's the NIO code:

https://github.com/apple/swift-nio/blob/6050c9c864d641f66114cfd6ab690f1aa51d74d4/Sources/NIO/ChannelInvoker.swift#L141-L144